### ISSUE TITLE
Rule blocked: CORERULES-249

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/cdisc-rules-engine#397]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/browse/CORERULES-249
- Copied from: https://dev.azure.com/cdisc-org/cdisc-core/_workitems/edit/4757

**Rule Information**
- **Standard**: SDTM
- **Rule ID**: CG0562

**Describe the problem**
There needs to be a way to flatten lists of values in the `is_unique_set` and `is_not_unique_set` operations.

**Proposed rule logic**
```yaml
Check:
  all:
    - name: --REPNUM
      operator: exists
    - name: --REPNUM
      operator: is_not_unique_set
      value:
         - USUBJID
         - --TESTCD
         - $TIMING_VARIABLES
Operations:
  - id: $TIMING_VARIABLES
    name: Role
    value: Timing
    operator: library_variables
```

In this example, we need `value` to represent a flat list containing `USUBJID`, `--TESTCD`, and all of the values within the `$TIMING_VARIABLES` reference.

**Comment by RamilCDISC** [comment]
The validation process for the issue is complete. The acceptance criteria is properly implemented. The changes were tested through running unittests. The tests check the edge cases verifying by passing nested lists in the comparator of is_not_unique_set function. The nested lists are flattened successfully and the operation works fine. This fulfills the AC.

See the attached screenshots for reference

<img width="1201" alt="397" src="https://github.com/cdisc-org/cdisc-rules-engine/assets/113539111/2284e166-ec4d-4356-b199-a89ff5fd48d4">

**Comment by RamilCDISC** [comment]
### Validation Summary:

### Validation Methodology:
The validation process incorporated rigorous testing methodologies to ensure the accuracy and reliability of the system. This included:
**Unit Testing**:
Unit Testing was done to verify the implementation of the Acceptance criteria. Unit tests were designed to test the made changes using edge cases. All those tests were then run using pytest to verify compliance with acceptance criteria.

**Test Endpoint:** 
A rule was written using the logic provided in the PR related to this issue and then tested against dataset files having negative cases regarding the implementation. 
**Code Review:**
The code changes in the code files were reviewed and were found to be in accordance with the Acceptance criteria. Also, the code was reviewed for following: 
Code Quality Evaluation
Logic and Functionality Review
Error Handling and Exception Handling Assessment
Security Considerations
**Regression Testing:**
Regression tests were run to validate that the changes did not affect any functionality of the overall system.

### Test Results:
**Test Execution:**
The tests were executed in the following order:

> **Code Review:** The code for all the changes files was reviewed first to check the proper implementation of the Acceptance criteria
>     
> **Unit Tests:** The unit tests were run using the following command in the terminal from the root of the project
> `python -m pytests path/to/the/test/files`
> **Test Endpoint**: Following rule logic was written to test the implementation using the endpoint
>     
>             Check:
>               all:
>                 - name: --REPNUM
>                   operator: exists
>                 - name: --REPNUM
>                   operator: is_not_unique_set
>                   value:
>                      - USUBJID
>                      - --TESTCD
>                      - $TIMING_VARIABLES
>             Operations:
>               - id: $TIMING_VARIABLES
>                 name: Role
>                 value: Timing
>                 operator: library_variables.
>             
> **Regression Tests**

  
  **Results:**
  After successful running of unit, Test endpoint rule, regression tests and careful validation, the issue was found to be good to close as the Acceptance criteria was fulfilled and all validations passed.

### Deviations and Non-conformances:
No deviations or non-conformances were observed.

### Summary of Findings:
**Test Results**: All tests conducted during the validation process passed successfully, indicating that the system functions as intended and meets the specified requirements. The tests covered various scenarios and edge cases, providing confidence in the system's reliability and correctness.
**Unit Regression:** The unit regression tests were executed, and no regressions or issues were identified. This indicates that the changes made during the development or modification of the system did not introduce any unintended side effects or break existing functionality.
**Code Reviews:** The code reviews conducted by both development and QA teams were completed, and no major issues or concerns were identified. The code adheres to coding standards and best practices, demonstrating good code quality, readability, and maintainability.
